doc/arch/adr-002-form-representation.md

```diff
@@ -0,0 +1,95 @@
+# ADR 2: JSON for form representation and storage
+
+## Context
+
+Bugle forms needs a data model to support these uses:
+- Users create forms with an arbitrary number of questions.
```

Comment:
A better phrasing would be: Users can create forms with _many_ questions. `Many` is well understood, `arbitrary` is less so. 

Similarly below: Users provide _many_ responses for each question. If the number of responses changes with the kind of question, consider outlining that in the ADR: What kind of question can have what number of responses?